akka-http-core/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala

```diff
@@ -163,4 +170,17 @@ private[http2] object RequestParsing {
     if (seenRegularHeader) malformedRequest(s"Pseudo-header field '$name' must not appear after a regular header")
   private[http2] def malformedRequest(msg: String): Nothing =
     throw new RuntimeException(s"Malformed request: $msg")
+  private[http2] def validateHeader(httpHeader: HttpHeader) = httpHeader.lowercaseName match {
+    case "connection" =>
+      // https://tools.ietf.org/html/rfc7540#section-8.1.2.2
+      malformedRequest("Header 'Connection' must not be used with HTTP/2")
+    case "transfer-encoding" =>
+      // https://tools.ietf.org/html/rfc7540#section-8.1.2.2
+      malformedRequest("Header 'Transfer-Encoding' must not be used with HTTP/2")
+    case "te" =>
+      // https://tools.ietf.org/html/rfc7540#section-8.1.2.2
+      if (httpHeader.value.compareToIgnoreCase("trailers") != 0) malformedRequest(s"Header 'TE' must not contain value other than 'trailers', value was '${httpHeader.value}")
```

Comment:
I think TE is used as short for Transfer-Encoding in the spec here, so:

```suggestion
    case "transfer-encoding" =>
      // https://tools.ietf.org/html/rfc7540#section-8.1.2.2
      if (httpHeader.value.compareToIgnoreCase("trailers") != 0) malformedRequest(s"Header 'TE' must not contain value other than 'trailers', value was '${httpHeader.value}")
```

Reply:
I may be wrong but I think it actually is a distinct header: https://tools.ietf.org/html/rfc7230#section-4.3

Reply:
huh, looks like you're completely right, sorry about he noise :(

Reply:
No worries, it sure is confusing with two headers meaning the same but not quite the same...